Sming/system/include/esp_systemapi.h

```diff
@@ -38,7 +39,14 @@
 #ifdef SMING_RELEASE
 #define debugf(fmt, ...)
 #else
-#define debugf(fmt, ...) m_printf(fmt"\r\n", ##__VA_ARGS__)
+//(!)Important: Normally debugf is defined as debug_e.
+//There are rare cases when expanding debug_e will cause a section type conflict
+// this is most likely because you are using debugf in a virtual member
+// of a class and a member of another class in the same file.
+//
+//If you get a section type conflict, please use m_printf instead of debugf
+// which will store the format string in RAM.
+#define debugf debug_e
```

Comment:
I was planning another 2 PRs
1. replace all debugf with corresponding d/i/w/e functions. Ex in Websocket code there are some debugf commented (debug) and the rest of them some are not useful. Some are errors that should stay.
2. modify debug function in axtls and in open_lwip -> they also take up a lot of RAM